lib/lib.go

```diff
@@ -127,22 +127,17 @@ func LoadConfig(ctx context.Context) (*FileConfig, error) {
 		f, err = os.Open(filename)
 		checkedLocations[0] = filename
 	} else {
-		homeDir, err := os.UserHomeDir()
+		var homeDir string
+		homeDir, err = os.UserHomeDir()
 		if err != nil {
 			return nil, err
 		}
 		filename = filepath.Join(homeDir, "cfg", "buildkite")
 		f, err = os.Open(filename)
-		if err != nil {
-			return nil, err
-		}
 		checkedLocations[0] = filename
 		if err != nil { // fallback
 			rcFilename := filepath.Join(homeDir, ".buildkite")
 			f, err = os.Open(rcFilename)
-			if err != nil {
-				return nil, err
```

Comment:
Hmmm, I think if both of these checks fail, we can fall through to `f.SetDeadline()`, where `f` might be nil and cause a NPE.